utils/esr_peer_info.cpp

```diff
@@ -408,14 +410,19 @@ namespace
         static const auto PEER_INFO_ABI{str_to_json(GET_PEER_INFO_ABI)};
         static const auto GET_PEER_INFO_SIGNATURE{PEER_INFO_ABI["signature"].asCString() + 2};
 
+        static const std::string PARAMS
+                {
+                    size_type_to_hex(swarm_id.size(), 64)             // size of swarm id string (pre hexification)
+                    + pad_str_to_mod_64(string_to_hex(swarm_id))    // parameter 1 - swarm id
+                    + size_type_to_hex(peer_id.size(), 64)          // size of peer id (pre hexification)
+                    + pad_str_to_mod_64(string_to_hex(peer_id))     // parameter 2 - peer id
+                };
+
         return std::string{"0x"
             + pad_str_to_mod_64(GET_PEER_INFO_SIGNATURE)
             + pad_str_to_mod_64("00000040")                 // first param type?
-            + pad_str_to_mod_64("00000080")                 // second param type?
-            + size_type_to_hex(swarm_id.size())             // size of swarm id string (pre hexification)
-            + pad_str_to_mod_64(string_to_hex(swarm_id))    // parameter 1 - swarm id
-            + size_type_to_hex(peer_id.size(), 64)          // size of peer id (pre hexification)
-            + pad_str_to_mod_64(string_to_hex(peer_id))     // parameter 2 - peer id
+            + size_type_to_hex(PARAMS.size() / 2)           // size of params blob in bytes, padded to 8 chars
```

Comment:
I _think_ there's an issue here if I'm understanding correctly. This will always indicate the start of the value for the second parameter (`peer_id`) is always halfway past the size of the encoding of _both_ parameters. Since these two strings can vary in size independently, this will not always be accurate.

Since the first parameter is a string, 32 bytes are used to indicate the offset to find the encoding for it. As the first parameter, it will be at the end of the parameter list. The second and last parameter is also a string, so another 32 bytes. The real data therefore begins 64 bytes later, which is where the `00000040` (or 0x40, or 64) comes from in the first bit. The mod64 is of course because hex chars are half bytes. The first string will always start here.

The next 32 bytes needs to indicate the offset for the second parameter's encoding. The varies based on the encoding (32 byte length, n*32 byte data value) of the first parameter, and only the first parameter. `PARAMS.size() / 2` accounts for the size of both parameters, which is incorrect.